backend/src/ncpdpScriptBuilder/buildScript.v2017071.js

```diff
@@ -46,14 +54,130 @@ export default function buildRxStatus(newOrder) {
       }
     ]
   };
-  const options = {
-    ignoreAttributes: false,
-    attributeNamePrefix: '@@',
-    format: true,
-    oneListGroup: 'true'
-  };
-  const builder = new XMLBuilder(options);
-  var RxStatus = builder.build(rxStatus);
 
-  return RxStatus;
+  const builder = new XMLBuilder(XML_BUILDER_OPTIONS);
+  return builder.build(rxStatus);
 }
+
+export const buildRxFill = newRx => {
+  const { Message } = JSON.parse(newRx.serializedJSON);
+  const { Header, Body } = Message;
+  const time = new Date();
+  const message = {
+    Message: {
+      Header: [
+        {
+          To: {
+            '#text': Header.From._,
+            '@@Qualifier': Header.From.$.Qualifier
+          }
+        },
+        {
+          From: {
+            '#text': Header.To._,
+            '@@Qualifier': Header.To.$.Qualifier
+          }
+        },
+        { MessageID: uuidv4() },
+        { RelatesToMessageID: Header.MessageID },
+        { SentTime: time.toISOString() },
+        { PrescriberOrderNumber: Header.PrescriberOrderNumber }
+      ],
+      Body: [
+        {
+          RxFill: {
+            FillStatus: {
+              Dispensed: {
+                Note: PICKED_UP
+              }
+            },
+            Patient: {
+              HumanPatient: {
+                Name: {
+                  LastName: Body.NewRx.Patient.HumanPatient.Name.LastName,
+                  FirstName: Body.NewRx.Patient.HumanPatient.Name.FirstName
+                },
+                Gender: Body.NewRx.Patient.HumanPatient.Gender,
+                DateOfBirth: { Date: Body.NewRx.Patient.HumanPatient.DateOfBirth.Date },
+                Address: {
+                  AddressLine1: Body.NewRx.Patient.HumanPatient.Address.AddressLine1,
+                  City: Body.NewRx.Patient.HumanPatient.Address.City,
+                  StateProvince: Body.NewRx.Patient.HumanPatient.Address.StateProvince,
+                  PostalCode: Body.NewRx.Patient.HumanPatient.Address.PostalCode,
+                  Country: Body.NewRx.Patient.HumanPatient.Address.Country
+                }
+              }
+            },
+            Pharmacy: {
+              Identification: {
+                NCPDPID: MOCK_VALUE,
```

Comment:
What is MOCK_VALUE? should these all be the same?

Reply:
MOCK_VALUE is just a dummy string 'MOCK_VALUE'. I wasn't sure what to put for the Pharmacy information- unless I should put in some more convincing hardcoded strings? Although I did forget to pass the pharmacy name over from the NewRx addressing it as "Pharmacy 123".

Reply:
It would be nice if there was some reasonable fake data in here just like our Patients are all fake data.